packages/@lwc/jest-serializer/src/clean-element-styles.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+function cleanElementStyles(elm) {
+    // Normalize trivial whitespace differences (e.g. `style="  foo   bar "` becomes `style="foo bar"`
+    // and `style=""` is just removed).
+    const styleAttribute = elm.getAttribute('style') || '';
+    const normalizedStyleAttribute = styleAttribute
+        // Remove leading/trailing space
+        .trim()
+        // Replace multiple space, tab, line break with single space
+        .replace(/[\s\n\t]+/g, ' ')
+        .split(';')
+        // Preserve space character following a ';', style="color: blue; text-align: center"
+        .map((attr) => attr.trimEnd())
+        .join(';');
```

Comment:
```suggestion
        .split(';')
        .map(_ => _.trim())
        .filter(Boolean)
        .map(_ => `${_};`)
        .join(' ');
```

This avoids the duplicates `;`s at the end.

Reply:
```suggestion
        .split(';')
        .map(_ => `${_.trim()};`)
        .filter(_ => _ !== ';')
        .join(' ');
```

Same but one less loop.

Reply:
This will also append a ';' at the end of a style:

```html
<div style="  color: blue  "></div>
```

Becomes:

```html
<div style="color: blue;"></div>
```

Empty ';' will be removed as well

```html
<div style=";;"></div>
```
 Becomes:

```html
<div />
```

Just want to double-check, are we trying to normalize the whitespace or the style attribute output?

Asking because it won't exactly line up with what the component author will see in the browser.

Reply:
Yes, I think we want consistent output. So consistent `;`s at the end, consistent spacing, and just `<div />` if the `style` is semantically empty.